### PR TITLE
Build dynamic type library on more standards

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

File: .github/workflows/lint.yml [hunk 1]
```diff
@@ -41,7 +41,7 @@ jobs:
         working-directory: ${{ env.working_directory }}
         run: |
           sudo apt update && sudo apt install -y libgtest-dev libgmock-dev libbenchmark-dev &
-          pip install meson ninja &
+          pip install git+https://github.com/mesonbuild/meson ninja &
```
Comment (zasdfgbnm) [inline .github/workflows/lint.yml:44]: Using nightly meson, because the latest release of meson doesn't know that gcc 11 supports `-std=c++23`.

**Comment by zasdfgbnm** [comment]
!build

**Comment by zasdfgbnm** [comment]
I believe all our supported platforms supports `-std=c++23`, although very few C++23 features might have been implemented. It is never wrong to build and test on as much standards as possible.

@naoyam Could you check if this breaks your environment with clang?

**Comment by naoyam** [comment]
> I believe all our supported platforms supports `-std=c++23`, although very few C++23 features might have been implemented. It is never wrong to build and test on as much standards as possible.
> 
> @naoyam Could you check if this breaks your environment with clang?

clang-14 on Ubuntu 22.04 works fine.

**Comment by zasdfgbnm** [comment]
> clang-14 on Ubuntu 22.04 works fine.

Thanks for checking! Let's merge https://github.com/NVIDIA/Fuser/pull/1076 first, and rerun CI here to double check.

**Comment by zasdfgbnm** [comment]
> > clang-14 on Ubuntu 22.04 works fine.
> 
> Thanks for checking! Let's merge #1076 first, and rerun CI here to double check.

Done

**Review by naoyam — APPROVED** [review]
Stamping.

**Pull request merged**